hbase-external-blockcache/src/main/java/org/apache/hadoop/hbase/io/hfile/MemcachedBlockCache.java

```diff
@@ -67,10 +78,20 @@ public class MemcachedBlockCache implements BlockCache {
   public static final String MEMCACHED_OPTIMIZE_KEY = "hbase.cache.memcached.spy.optimze";
   public static final long MEMCACHED_DEFAULT_TIMEOUT = 500;
   public static final boolean MEMCACHED_OPTIMIZE_DEFAULT = false;
+  public static final int STAT_THREAD_PERIOD = 60 * 5;
 
   private final MemcachedClient client;
   private final HFileBlockTranscoder tc = new HFileBlockTranscoder();
   private final CacheStats cacheStats = new CacheStats("MemcachedBlockCache");
+  private final AtomicLong cachedCount = new AtomicLong();
+  private final AtomicLong notCachedCount = new AtomicLong();
+  private final AtomicLong cacheErrorCount = new AtomicLong();
+  private final AtomicLong timeoutCount = new AtomicLong();
+
+  /** Statistics thread schedule pool (for heavy debugging, could remove) */
+  private transient final ScheduledExecutorService scheduleThreadPool =
+    Executors.newScheduledThreadPool(1, new ThreadFactoryBuilder()
+      .setNameFormat("MemcachedBlockCacheStatsExecutor").setDaemon(true).build());
```

Comment:
Since it is meant for debugging, shall we initialize the thread pool with only some configuration (default off)?

Reply:
What do you have in mind? This is the same as Bucket Cache.

Reply:
We want to log stats every 5 minutes from this cache just like we do with bucket cache.